src/Cassandra.Tests/SerializerTests.cs

```diff
@@ -87,61 +87,139 @@ public void EncodeDecodeSingleValuesTest()
         /// Tests that the default target type when is not provided
         /// </summary>
         [Test]
-        public void EncodeDecodeSingleValuesDefaultsFactory()
+        [TestCaseSource(nameof(SingleValuesTestCases))]
+        public void EncodeDecodeSingleValuesDefaultsFactory(object[] feed)
         {
-            var initialValues = new object[]
+            var value = ((Array)feed[0]).GetValue(0);
+            var code = (ColumnTypeCode)feed[1];
+            foreach (var protocolVersion in _protocolVersions)
+            {
+                var serializer = NewInstance(protocolVersion);
+                byte[] encoded = serializer.Serialize(value);
+                //Set object as the target CSharp type, it should get the default value
+                Assert.AreEqual(value, serializer.Deserialize(encoded, code, null));
+            }
+        }
+
+        //TODO: exhaust the types
```

Comment:
Adding the following would exhaust the types and remove the TODO comment
```java
            yield return new object[] { new sbyte[] { 0, 1 }, ColumnTypeCode.TinyInt };
yield return new object[] { new short[] { -1, 1 }, ColumnTypeCode.SmallInt };
            yield return new object[] { new[] { BigInteger.Parse("10000000"), BigInteger.One }, ColumnTypeCode.Varint };
            yield return new object[] { new[] { Duration.Zero, Duration.Parse("1y2mo") }, ColumnTypeCode.Duration };
            yield return new object[] { new[] { new LocalDate(2020, 1, 2), new LocalDate(1970, 12, 12) }, ColumnTypeCode.Date };
            yield return new object[] { new[] { new LocalTime(1, 2, 3, 4), new LocalTime(23, 59, 59, 999) }, ColumnTypeCode.Time };
```

Reply:
Awesome, thanks!